### PR TITLE
hdl.ast: fix Array not being indexable by ValueCastable

### PR DESCRIPTION
**Pull request by anuejn**

### CODE REVIEW

File: amaranth/hdl/ast.py [hunk 1]
```diff
@@ -1382,6 +1382,8 @@ def __init__(self, iterable=()):
         self._mutable  = True
 
     def __getitem__(self, index):
+        if isinstance(index, ValueCastable):
```
Comment (whitequark) [inline amaranth/hdl/ast.py:1385]: You don't need this check.

**Review by whitequark — CHANGES_REQUESTED** [review]
You should add a test.

File: amaranth/hdl/ast.py (hunk 1 above)
Comment (anuejn) [reply amaranth/hdl/ast.py:1385]: If we remove this check, we change the semantics of the method. currently if we give the method an int, we will not get an ArrayProxy. With this change, we would.

Comment (whitequark) [reply amaranth/hdl/ast.py:1385]: Oh, you are right.

**Review by whitequark — APPROVED** [review]

**Pull request merged**